src/dwarf_parser.cpp

```diff
@@ -295,17 +295,19 @@ std::vector<std::string> Dwarf::get_function_params(
   return result;
 }
 
-ProbeArgs Dwarf::resolve_args(const std::string &function)
+Struct Dwarf::resolve_args(const std::string &function)
 {
-  std::map<std::string, SizedType> result;
+  Struct result(0, false);
   int i = 0;
   for (auto &param_die : function_param_dies(function))
   {
     Dwarf_Die type_die = type_of(param_die);
     SizedType arg_type = get_stype(type_die);
     arg_type.is_funcarg = true;
     arg_type.funcarg_idx = i++;
-    result.emplace(dwarf_diename(&param_die), arg_type);
+    result.AddField(
+        dwarf_diename(&param_die), arg_type, result.size, std::nullopt, false);
+    result.size += arg_type.GetSize();
```

Comment:
Not at this moment. The struct is never built (it's just a placeholder type) and the individual args are read from the registers. Once I rebase the support for working with the entire `args` for uprobes (#2477), the struct will be built on stack, so this will matter. Still, I'll try to build the struct without padding to save stack space.